OADP-4058: Adding structure for OADP and 3scale user story (#85994)
openshift/openshift-docs: modules/scale-up-the-3scale-operator-and-deployment.adoc

:_mod-docs-content-type: PROCEDURE

//included in backing-up-and-restoring-3scale-using-oadp.adoc assembly

[id="scale-up-the-3scale-operator-and-deployment_{context}"]

> = Scale up the 3scale operator and deployment

Reviewer: ```suggestion
= Scaling up the 3scale operator and deployment
```
The name should contain the -ing verb as per our rules for procedures.

Author: Good catch, missed adding a gerund for this module :)

> Scale up the 3scale operator and any deployment that were manually scaled down. After a few minutes, 3scale installation should be fully functional, and its state should match the backed-up state.

Reviewer: ```suggestion
You can scale up the 3scale operator and any deployment that was manually scaled down. After a few minutes, 3scale installation should be fully functional, and its state should match the backed-up state.
```

.Prerequisites

> * Ensure that there are no deployments scaled up already, or any extra pods are running. 

Reviewer: ```suggestion
* Ensure that there are no scaled up deployments or no extra pods are running. 
```

There might be some system-mysql or backend-redis pods running detached from deployments after restoration, which can be removed after the restoration is successful.

.Procedure

. Scale up the 3scale Operator by executing the following command:
+
[source,terminal]
----
$ oc scale deployment threescale-operator-controller-manager-v2 --replicas=1 -n threescale
----

. Ensure if the 3scale operator was deployed: 
+
[source,terminal]
----
$ oc get deployment -n threescale
----

> . Scale up the deployments using the following script:

Reviewer: ```suggestion
. Scale up the deployments by running the following script:
```

+
[source,terminal]
----
$ cat ./scaledeployment.sh

for dc in apicast-production apicast-staging backend-cron backend-listener backend-redis backend-worker system-app system-memcache system-mysql system-redis system-searchd system-sidekiq zync zync-database zync-que; do
    oc scale dc/$dc --replicas=1 -n threescale
Done
----

. Execute the script:
+
[source,terminal]
----
$ ./scaledeployment.sh
----

> . Get the `3scale-admin` route to log in to the 3scale UI by running the following command:

Reviewer: ```suggestion
. Get the `3scale-admin` route to log in to the 3scale UI:
```
